src/tiled/commandmanager.cpp

```diff
@@ -59,17 +59,32 @@ CommandDataModel *CommandManager::commandDataModel()
     return mModel;
 }
 
+void CommandManager::registerMenu(QMenu *menu)
+{
+    mMenus.append(menu);
+    menu->clear();
+    menu->addActions(mActions);
+}
+
 void CommandManager::showDialog()
 {
     CommandDialog dialog(QApplication::activeWindow());
     dialog.exec();
 }
 
-void CommandManager::populateMenu(QMenu *menu, bool flag)
+void CommandManager::populateMenus()
 {
-    menu->clear();
+    for (int i = 0; i < mMenus.size(); ++i) {
+        QMenu *menu = mMenus.at(i);
```

Comment:
You were not using indexes? `updateActions` does use the index now at least, in the following line:
```c++
connect(mAction, &QAction::triggered, [this,i]() { mModel->execute(i); });
```